File: backend/app/schemas/catalog.py
"""Schemas para ClubTag y ClubCatalogEntry."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.drill import DrillSummaryResponse


class ClubTagCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    color: str | None = Field(None, pattern=r"^#[0-9a-fA-F]{6}$")


class ClubTagUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=100)
    color: str | None = Field(None, pattern=r"^#[0-9a-fA-F]{6}$")


class ClubTagResponse(BaseModel):
    id: int
    club_id: int
    name: str
    color: str | None
    archived_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class PublishToCatalogRequest(BaseModel):
    drill_id: int
    tag_ids: list[int] = Field(default_factory=list)


class UpdateCatalogTagsRequest(BaseModel):
    """Actualiza los tags del club asociados a una entrada del catálogo."""
    tag_ids: list[int] = Field(default_factory=list)


class CatalogEntryResponse(BaseModel):
    id: int
    club_id: int
    drill: DrillSummaryResponse
    original_drill_id: int | None
    published_by: int
    tags: list[ClubTagResponse]
    archived_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: shared/types/catalog.ts
import type { DrillSummary } from "./drill";

// ── ClubTag ───────────────────────────────────────────────────────────────────

export interface ClubTag {
  id: number;
  club_id: number;
  name: string;
  color: string | null;
  archived_at: string | null;
  created_at: string;
}

export interface ClubTagCreate {
  name: string;
  color?: string | null;
}

export interface ClubTagUpdate {
  name?: string;
  color?: string | null;
}

// ── CatalogEntry ──────────────────────────────────────────────────────────────

export interface CatalogEntry {
  id: number;
  club_id: number;
  drill: DrillSummary;
  original_drill_id: number | null;
  published_by: number;
  tags: ClubTag[];
  archived_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface PublishToCatalogRequest {
  drill_id: number;
  tag_ids?: number[];
}

export interface UpdateCatalogTagsRequest {
  tag_ids: number[];
}

// ── TeamPlaybookEntry ─────────────────────────────────────────────────────────

export interface PlaybookEntry {
  id: number;
  team_id: number;
  drill: DrillSummary;
  added_by: number;
  is_frozen: boolean;
  frozen_at: string | null;
  archived_at: string | null;
  /** Coach annotation visible to all team members. */
  note: string | null;
  created_at: string;
}

export interface AddToPlaybookRequest {
  drill_id: number;
}
